src/ports/postgres/modules/stats/pred_metrics.sql_in

```diff
@@ -0,0 +1,831 @@
+<DT>grouping_cols (optional)</DT>
+<DD>TEXT, default: NULL. Name of the column of grouping values from input
+table.</DD>
+</DL>
+
+@anchor specs
+@par Function Specific Details
+
+<b>R-squared Score</b>
+
+This function returns the coefficient of determination (R2) between the
+predicted and observed values. An R2 of 1 indicates that the regression line
+perfectly fits the data, while an R2 of 0 indicates that the line does not fit
+the data at all. Negative values of R2 may occur when fitting non-linear
+functions to data. Please refer to the reference <a href="#r2">[1]</a> for
+further details.
+
+<b>Adjusted R-squared Score</b>
+
+This function returns the adjusted R2 score. Adjusted R2 score is used to
+counter the problem of the R2 automatically increasing when extra explanatory
+variables are added to the model. It takes additional two integers describing
+the degrees of freedom of the model and the size of the training set over which
+it was developed, and returning the adjusted R-squared prediction accuracy
+metric. Please refer to the reference <a href="#r2">[1]</a> for further details.
+
+Arguments:
+
+- num_predictors: Indicates the number of parameters the model has other than
+the constant term. For example, if it is set to '3' the model may take the
+following form, 7 + 5x + 39y + 0.91z.
+- training_size: Indicates the number of rows in the training set (excluding
+any NULL rows).
+
+Neither of these arguments can be deduced from the predicted values and the test data alone.
+
+@anchor bc
+<b>Binary Classification</b>
+
+This function returns an output table with a number of metrics commonly used in binary classification.
+
+The definitions of the various metrics are as follows.
+
+- \f$\textit{tp}\f$ is the count of correctly-classified positives.
+- \f$\textit{tn}\f$ is the count of correctly-classified negatives.
+- \f$\textit{fp}\f$ is the count of misclassified negatives.
+- \f$\textit{fn}\f$ is the count of misclassified positives.
+- \f$\textit{tpr}=\textit{tp}/(\textit{tp}+\textit{fn})\f$.
+- \f$\textit{tnr}=\textit{tn}/(\textit{fp}+\textit{tn})\f$.
+- \f$\textit{ppv}=\textit{tp}/(\textit{tp}+\textit{fp})\f$.
+- \f$\textit{npv}=\textit{tn}/(\textit{tn}+\textit{fn})\f$.
+- \f$\textit{fpr}=\textit{fp}/(\textit{fp}+\textit{tn})\f$.
+- \f$\textit{fdr}=1-\textit{ppv}\f$.
+- \f$\textit{fnr}=\textit{fn}/(\textit{fn}+\textit{tp})\f$.
+- \f$\textit{acc}=(\textit{tp}+\textit{tn})/(\textit{tp}+\textit{tn}+\textit{fp}+\textit{fn})\f$.
+- \f$\textit{f1}=2*\textit{tp}/(2*\textit{tp}+\textit{fp}+\textit{fn})\f$.
+
+
+<b>Area under ROC Curve</b>
+
+This function returns the area under the Receiver Operating Characteristic curve
+for binary classification (the AUC). The ROC curve is the curve relating the
+classifier's TPR and FPR metrics. (See <a href="#bc">Binary Classification</a>
+for a definition of these metrics). Please refer to the reference <a
+href="#aoc">[2]</a> for further details. Note that the binary classification
```

Comment:
Please check if the reference number is correct. [2] points to confusion matrix.